cli/app/commands/conflict/conflict.py

```diff
@@ -0,0 +1,391 @@
+import os
+import subprocess
+import re
+import yaml
+from typing import Dict, List, Optional, Any, Tuple
+from packaging import version
+from packaging.specifiers import SpecifierSet
+from packaging.version import Version
+
+from app.utils.logger import Logger
+from app.utils.protocols import LoggerProtocol
+from app.utils.output_formatter import OutputFormatter
+from app.utils.lib import ParallelProcessor
+from .models import ConflictCheckResult, ConflictConfig
+from .messages import (
+    error_checking_tool_version,
+    error_parsing_version,
+    timeout_checking_tool,
+    tool_not_found,
+    tool_version_mismatch,
+    tool_version_compatible,
+    conflict_config_not_found,
+    conflict_invalid_config,
+    conflict_loading_config,
+    conflict_config_loaded,
+)
+
+
+class VersionParser:
+    """Utility class for parsing and comparing versions."""
+    
+    # Version pattern mappings for different tools
+    VERSION_PATTERNS = [
+        r"version\s+(\d+\.\d+\.\d+)",
+        r"v(\d+\.\d+\.\d+)",
+        r"(\d+\.\d+\.\d+)",
+        r"Version\s+(\d+\.\d+\.\d+)",
+        r"(\d+\.\d+)",
+    ]
+    
+    @staticmethod
+    def parse_version_output(tool: str, output: str) -> Optional[str]:
+        """Parse version from tool output."""
+        try:
+            # Common version patterns
```

Comment:
what is meant by common version patterns here?

Reply:
Common are those cases where you get version values directly on upon giving command where as some are very specific to the tool, they dont provide the version directly. This is not just go, but curl, psql etc.. have handled it for those.